erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -506,6 +549,18 @@ def _validate_work_order(pro_doc):
 				if self.purpose == "Manufacture":
 					pro_doc.run_method("update_planned_qty")
 
+			if (self.purpose == "Material Consumption for Manufacture" or self.purpose == "Manufacture"):
+				for items in self.items:
+					if self.docstatus == 1:
+						frappe.db.sql("""Update `tabWork Order Item` set consumed_qty = consumed_qty + %s
```

Comment:
Always avoid codes like this which does "+" and "- for updating qty. Use `sum` of all consumed qty against the work order, similar to https://github.com/frappe/erpnext/blob/develop/erpnext/manufacturing/doctype/work_order/work_order.py#L498